source/hp/fe_collection.cc

```diff
@@ -103,6 +109,69 @@ namespace hp
 
 
 
+  template <int dim, int spacedim>
+  unsigned int
+  FECollection<dim, spacedim>::find_most_face_dominating_fe_in_subset(
+    const std::set<unsigned int> &fes) const
+  {
+    const hp::FECollection<dim, spacedim> &fe_collection = *this;
+
+    // If the set of elements to be dominated contains only a single element X,
+    // then by definition the dominating set contains this single element
+    // (because each element can dominate itself).
+    // There may also be others, in which case we'll check if any of these
+    // elements is able to dominate all others. If one was found, we stop
+    // looking further and return the dominating element.
+    if (fes.size() == 1)
+      {
+        Assert(*fes.begin() < fe_collection.size(),
+               ExcIndexRangeType<unsigned int>(*fes.begin(),
+                                               0,
+                                               fe_collection.size()));
+
+        return *fes.begin();
+      }
+
+    // first loop over all FEs and check which can dominate those given in @p fes:
```

Comment:
wait a second, aren't you looking for dominating within `@p fes` only? the rest of FECollection should be irrelevant?

Reply:
what I meant is that you don't loop over **all**, but the code actually loop over `@p fes`. So I think comment needs adjustment.